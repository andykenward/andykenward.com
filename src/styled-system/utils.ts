import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { colorPropsKeys } from "./colors"
import { fontSizePropsKeys } from "./fonts"

const propKeys = [...colorPropsKeys, ...fontSizePropsKeys]

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)
