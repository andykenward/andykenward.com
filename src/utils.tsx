import shouldForwardProp from "@styled-system/should-forward-prop"
import { ThemeProvider } from "emotion-theming"
import { equals, indexOf } from "ramda"
import React from "react"
import { render } from "@testing-library/react"

import { theme } from "./theme"

export const filterProps: (prop: string, propsToFilter: string[]) => boolean = (
  prop: string,
  propsToFilter: string[]
): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propsToFilter), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" || prop === "data-test-id"

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>)
