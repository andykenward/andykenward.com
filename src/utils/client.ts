import { GraphQLClient } from "graphql-request"

import { getSdk } from "../../generated/graphql"

const graphqlClient = new GraphQLClient(process.env.API_URL, {
  headers: {
    authorization: `Bearer ${process.env.API_TOKEN}`,
  },
})

export const appSdk = (() => getSdk(graphqlClient))()
