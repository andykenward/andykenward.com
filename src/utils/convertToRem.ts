import { is } from "ramda"
/**
 *
 * UTILS
 */

export const convertToRem = (n: number | string): string | number =>
  is(Number, n) && n != 0 ? `${(n as number) / 16}rem` : n
