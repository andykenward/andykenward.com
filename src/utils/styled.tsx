import styled, { CreateStyled } from "@emotion/styled"
import React from "react"

import {
  CSSObject,
  SerializedStyles,
  InterpolationWithTheme,
} from "@emotion/core"

import { Theme } from "../theme"

/**
 * TODO: pass `as` typings down if component
 */
export interface StyledProps {
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
  css?: CSSObject | SerializedStyles | InterpolationWithTheme<any>
}

export default styled as CreateStyled<Theme>
