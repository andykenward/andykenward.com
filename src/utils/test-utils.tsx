import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { ThemeProvider } from "theme-ui"

import { theme } from "../theme"

const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllTheProviders, ...options })

export * from "@testing-library/react"

export { customRender as render }
