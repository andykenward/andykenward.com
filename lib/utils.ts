import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { propKeys } from "./styled-system"

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" ?? prop === "data-test-id"
