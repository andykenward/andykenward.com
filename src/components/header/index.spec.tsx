import React from "react"

import { renderWithTheme } from "../../utils"

import Header from "."

describe("<Header />", () => {
  it("renders null when data is null", () => {
    const { container } = renderWithTheme(<Header data={null} />)
    expect(container.firstChild).toBeNull()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders null when data is undefined", () => {
    const { container } = renderWithTheme(<Header data={undefined} />)
    expect(container.firstChild).toBeNull()
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders when name is `null`", () => {
    const { container } = renderWithTheme(
      <Header data={{ name: null, role: "Role" }} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders when role is `null`", () => {
    const { container } = renderWithTheme(
      <Header data={{ name: "Name", role: null }} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const mockData = { name: "Andy Kenward", role: "Senior Front-End Engineer" }
    const { container } = renderWithTheme(<Header data={mockData} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
