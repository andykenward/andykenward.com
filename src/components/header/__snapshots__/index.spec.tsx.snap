// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
.emotion-6 {
  background-color: rgb(216,234,204);
  padding-top: 64px;
  padding-bottom: 64px;
}

@media screen and (min-width:40em) {
  .emotion-6 {
    padding-top: 128px;
    padding-bottom: 128px;
  }
}

.emotion-5 {
  padding-left: 32px;
  padding-right: 32px;
}

@media screen and (min-width:40em) {
  .emotion-5 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media screen and (min-width:40em) {
  .emotion-5 {
    width: 75%;
  }
}

.emotion-0 {
  font-family: "Cooper Light",Georgia,Times,"Times New Roman",serif;
  font-size: 48px;
  font-weight: 300;
  line-height: 1.090909091;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 64px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 80px;
  }
}

.emotion-1 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-size: 36px;
  font-weight: 400;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 8px;
}

@media screen and (min-width:40em) {
  .emotion-1 {
    font-size: 46px;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  margin-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

<header
  class="emotion-6"
>
  <div
    class="emotion-5"
  >
    <h1
      class="emotion-0"
    >
      Andy Kenward
    </h1>
    <h2
      class="emotion-1"
    >
      Senior Front-End Engineer
    </h2>
    <div
      class="emotion-4"
    >
      <a
        class="emotion-2"
        href="a@a.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Email
      </a>
      <a
        class="emotion-2"
        href="github"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </div>
  </div>
</header>
`;

exports[`<Header /> renders null when data is null 1`] = `null`;

exports[`<Header /> renders null when data is undefined 1`] = `null`;

exports[`<Header /> renders when name is \`null\` 1`] = `
.emotion-6 {
  background-color: rgb(216,234,204);
  padding-top: 64px;
  padding-bottom: 64px;
}

@media screen and (min-width:40em) {
  .emotion-6 {
    padding-top: 128px;
    padding-bottom: 128px;
  }
}

.emotion-5 {
  padding-left: 32px;
  padding-right: 32px;
}

@media screen and (min-width:40em) {
  .emotion-5 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media screen and (min-width:40em) {
  .emotion-5 {
    width: 75%;
  }
}

.emotion-0 {
  font-family: "Cooper Light",Georgia,Times,"Times New Roman",serif;
  font-size: 48px;
  font-weight: 300;
  line-height: 1.090909091;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 64px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 80px;
  }
}

.emotion-1 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-size: 36px;
  font-weight: 400;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 8px;
}

@media screen and (min-width:40em) {
  .emotion-1 {
    font-size: 46px;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  margin-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

<header
  class="emotion-6"
>
  <div
    class="emotion-5"
  >
    <h1
      class="emotion-0"
    />
    <h2
      class="emotion-1"
    >
      Role
    </h2>
    <div
      class="emotion-4"
    >
      <a
        class="emotion-2"
        href=""
        rel="noopener noreferrer"
        target="_blank"
      >
        Email
      </a>
      <a
        class="emotion-2"
        href=""
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </div>
  </div>
</header>
`;

exports[`<Header /> renders when role is \`null\` 1`] = `
.emotion-6 {
  background-color: rgb(216,234,204);
  padding-top: 64px;
  padding-bottom: 64px;
}

@media screen and (min-width:40em) {
  .emotion-6 {
    padding-top: 128px;
    padding-bottom: 128px;
  }
}

.emotion-5 {
  padding-left: 32px;
  padding-right: 32px;
}

@media screen and (min-width:40em) {
  .emotion-5 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media screen and (min-width:40em) {
  .emotion-5 {
    width: 75%;
  }
}

.emotion-0 {
  font-family: "Cooper Light",Georgia,Times,"Times New Roman",serif;
  font-size: 48px;
  font-weight: 300;
  line-height: 1.090909091;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 64px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 80px;
  }
}

.emotion-1 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-size: 36px;
  font-weight: 400;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 1.25;
  text-transform: capitalize;
  color: rgb(36,28,21);
  margin: 0;
  margin-bottom: 8px;
}

@media screen and (min-width:40em) {
  .emotion-1 {
    font-size: 46px;
  }
}

.emotion-2 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  margin-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

<header
  class="emotion-6"
>
  <div
    class="emotion-5"
  >
    <h1
      class="emotion-0"
    >
      Name
    </h1>
    <h2
      class="emotion-1"
    />
    <div
      class="emotion-4"
    >
      <a
        class="emotion-2"
        href=""
        rel="noopener noreferrer"
        target="_blank"
      >
        Email
      </a>
      <a
        class="emotion-2"
        href=""
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </div>
  </div>
</header>
`;
